pkg/decoder/decoder_test.go

```diff
@@ -7,102 +7,225 @@ package decoder
+	inChan <- NewInput([]byte("1. Hello world!\n"))
+	out = <-outChan
+	assert.Equal(t, "1. Hello world!", string(out.Content))
+	assert.Equal(t, "", d.lineBuffer.String())
+
+	// multiple single-line messages in one raw data
+	inChan <- NewInput([]byte("1. Hello world!\n2. How are you\n"))
+	out = <-outChan
+	assert.Equal(t, "1. Hello world!", string(out.Content))
+	out = <-outChan
+	assert.Equal(t, "2. How are you", string(out.Content))
+	assert.Equal(t, "", d.lineBuffer.String())
+
+	// two lines too big message in one raw data
+	inChan <- NewInput([]byte("12345678.\n" + strings.Repeat("a", contentLenLimit+10) + "\n"))
+	out = <-outChan
+	assert.Equal(t, 11+contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// two lines too big message in two raw data
+	inChan <- NewInput([]byte("12345678.\n"))
+	inChan <- NewInput([]byte(strings.Repeat("a", contentLenLimit+10) + "\n"))
+	out = <-outChan
+	assert.Equal(t, 11+contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// single-line too big message over two raw data
+	inChan <- NewInput([]byte(strings.Repeat("a", contentLenLimit)))
+	inChan <- NewInput([]byte(strings.Repeat("a", 10) + "\n"))
+	out = <-outChan
+	assert.Equal(t, contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// single-line too big message in one raw data
+	inChan <- NewInput([]byte(strings.Repeat("a", contentLenLimit+10) + "\n"))
+	out = <-outChan
+	assert.Equal(t, contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// message twice too big in one raw data
+	inChan <- NewInput([]byte(strings.Repeat("a", 2*contentLenLimit+10) + "\n"))
+	out = <-outChan
+	assert.Equal(t, contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, len(TRUNCATED)+contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// message twice too big over two raw data
+	inChan <- NewInput([]byte(strings.Repeat("a", contentLenLimit+5)))
+	inChan <- NewInput([]byte(strings.Repeat("a", contentLenLimit+5) + "\n"))
+	out = <-outChan
+	assert.Equal(t, len(TRUNCATED)+contentLenLimit, len(out.Content))
 	out = <-outChan
-	assert.Equal(t, reflect.TypeOf(out), reflect.TypeOf(message.NewStopMessage()))
+	assert.Equal(t, len(TRUNCATED)+contentLenLimit+len(TRUNCATED), len(out.Content))
+	out = <-outChan
+	assert.Equal(t, string(TRUNCATED)+strings.Repeat("a", 10), string(out.Content))
+
+	// pending message in one raw data
+	inChan <- NewInput([]byte(("1. Hello world!")))
+	timeout := time.NewTimer(1*time.Second + 1*time.Millisecond)
```

Comment:
what is this test testing? Also it'd be great to avoid sleeping for a second in a test. Is there a way to force a ticker tick?

Reply:
We're just making sure that the content stays in buffer if the payload did no finish with '\n', which can happen if the decoder consumes much faster than the tailer produces data